spec/lib/whitehall_importer/embed_body_references_spec.rb

```diff
@@ -12,31 +12,135 @@
       expect(govspeak_body).to eq("[Contact:#{content_id}]")
     end
 
-    it "converts Whitehall image syntax to Content Publisher syntax" do
-      govspeak_body = described_class.call(
-        body: "!!1 test !!2",
-        images: ["foo.png", "bar.jpg"],
-      )
+    context "embedded images" do
+      it "does not convert two Windows carridge returns" do
+        govspeak_body = described_class.call(
+          body: "\r\n\r\n!!1",
+          images: ["file.jpg"],
+        )
 
-      expect(govspeak_body).to eq("[Image:foo.png] test [Image:bar.jpg]")
-    end
+        expect(govspeak_body).to eq("\r\n\r\n[Image:file.jpg]")
+      end
 
-    it "removes any image markdown that doesn't resolve to an image" do
-      govspeak_body = described_class.call(
-        body: "Bar !!2 Baz",
-        images: ["foo.png"],
-      )
+      it "does not convert two Unix carridge returns" do
+        govspeak_body = described_class.call(
+          body: "\n\n!!1",
+          images: ["file.jpg"],
+        )
+
+        expect(govspeak_body).to eq("\n\n[Image:file.jpg]")
+      end
+
+      it "converts one Windows carridge return" do
+        govspeak_body = described_class.call(
+          body: "\r\n!!1",
+          images: ["file.jpg"],
+        )
+
+        expect(govspeak_body).to eq("\r\n\r\n[Image:file.jpg]")
+      end
+
+      it "converts one Unix carridge return" do
+        govspeak_body = described_class.call(
+          body: "\n!!1",
+          images: ["file.jpg"],
+        )
 
-      expect(govspeak_body).to eq("Bar  Baz")
+        expect(govspeak_body).to eq("\n\n[Image:file.jpg]")
+      end
+
+      it "converts no carridge returns" do
+        govspeak_body = described_class.call(
+          body: "!!1",
+          images: ["file.jpg"],
+        )
+
+        expect(govspeak_body).to eq("[Image:file.jpg]")
+      end
+
+      it "converts Whitehall image syntax to Content Publisher syntax" do
```

Comment:
Should this test be `it "ignores images that are neither start of the string or preceeded by new lines"` ?

Reply:
Yup, makes sense.  There's also one further down for Attachments - we'll handle that further down in your suggestions.